### PR TITLE
INTYGFV-14931: Added exended functionality to CertificateSignConfirmationFunctionImpl that checks if REPLACED certificates exists for person.

### PR DESCRIPTION
**Pull request by ChrilleSigerhed**
Refactored CertificateSignConfirmationFunctionImpl for better readability.

### CODE REVIEW

File: web/src/main/java/se/inera/intyg/webcert/web/service/facade/impl/certificatefunctions/CertificateSignConfirmationFunctionImpl.java [hunk 1]
```diff
@@ -50,13 +53,15 @@ public CertificateSignConfirmationFunctionImpl(UtkastService utkastService,
 
     @Override
     public Optional<ResourceLinkDTO> get(Certificate certificate, WebCertUser webCertUser) {
-        if (!authoritiesHelper.isFeatureActive(AuthoritiesConstants.FEATURE_UNIKT_INTYG, certificate.getMetadata().getType())) {
+        if (isUniqueFeatureActive(certificate)) {
```
Comment (mww-sopra-steria) [inline web/src/main/java/se/inera/intyg/webcert/web/service/facade/impl/certificatefunctions/CertificateSignConfirmationFunctionImpl.java:56]: Maybe need to rename the method or move the "negating" operand (!) to outside the method. Now it read like if the unique feature is active it shouldn't return any resource link.

File: web/src/main/java/se/inera/intyg/webcert/web/service/facade/impl/certificatefunctions/CertificateSignConfirmationFunctionImpl.java [hunk 2]
```diff
@@ -50,13 +53,15 @@ public CertificateSignConfirmationFunctionImpl(UtkastService utkastService,
...
-        final var personnummer = Personnummer.createPersonnummer(certificate.getMetadata().getPatient().getPersonId().getId()).get();
-        final var existingIntyg = utkastService.checkIfPersonHasExistingIntyg(personnummer, webCertUser, certificate.getMetadata().getId());
-        final var previousIntygMap = existingIntyg.getOrDefault(INTYG_INDICATOR, Collections.emptyMap());
-        if (previousIntygMap.containsKey(certificate.getMetadata().getType())) {
+        final var personnummer = getPersonnummer(certificate);
+        final var existingIntyg = getExistingIntyg(certificate, webCertUser, personnummer);
+        final var previousIntygMap = existingIntygWithStatus(existingIntyg, INTYG_INDICATOR);
+        final var previousUtkastMap = existingIntygWithStatus(existingIntyg, UTKAST_INDICATOR);
+        final var previousReplacedMap = existingIntygWithStatus(existingIntyg, ERSATT_INDICATOR);
+        if (previousIntygExists(certificate, previousIntygMap) && !previousIntygExists(certificate, previousReplacedMap)) {
```
Comment (mww-sopra-steria) [inline web/src/main/java/se/inera/intyg/webcert/web/service/facade/impl/certificatefunctions/CertificateSignConfirmationFunctionImpl.java:64]: Not sure that it makes it clearer this way... Maybe should consider refactoring...

Now you it read like previousIntygexists && !previousIntygExists... and you have to check the arguments to understand. 

Maybe better to keep previousIntygMap.containKey(certificate.getMetadata().getType()). You could maybe put the type in a local variable so you reduce the code.

File: web/src/main/java/se/inera/intyg/webcert/web/service/facade/impl/certificatefunctions/CertificateSignConfirmationFunctionImpl.java [hunk 3]
```diff
@@ -65,8 +70,7 @@ public Optional<ResourceLinkDTO> get(Certificate certificate, WebCertUser webCer
                     + " Det är därför inte möjligt att signera detta dödsbevis.",
                 true));
         }
-        final var previousUtkastMap = existingIntyg.getOrDefault(UTKAST_INDICATOR, Collections.emptyMap());
-        if (previousUtkastMap.containsKey(certificate.getMetadata().getType())) {
+        if (previousIntygExists(certificate, previousUtkastMap) || previousIntygExists(certificate, previousReplacedMap)) {
```
Comment (mww-sopra-steria) [inline web/src/main/java/se/inera/intyg/webcert/web/service/facade/impl/certificatefunctions/CertificateSignConfirmationFunctionImpl.java:73]: Same as above comment.

**Review by mww-sopra-steria — CHANGES_REQUESTED** [review]
Nice! Just a few code readability comments.

File: web/src/main/java/se/inera/intyg/webcert/web/service/facade/impl/certificatefunctions/CertificateSignConfirmationFunctionImpl.java (hunk 1 above)
Comment (ChrilleSigerhed) [reply web/src/main/java/se/inera/intyg/webcert/web/service/facade/impl/certificatefunctions/CertificateSignConfirmationFunctionImpl.java]: I agree. Changed to negating operande.

File: web/src/main/java/se/inera/intyg/webcert/web/service/facade/impl/certificatefunctions/CertificateSignConfirmationFunctionImpl.java (hunk 2 above)
Comment (ChrilleSigerhed) [reply web/src/main/java/se/inera/intyg/webcert/web/service/facade/impl/certificatefunctions/CertificateSignConfirmationFunctionImpl.java]: Great suggestion! Updated.

**Review by mww-sopra-steria — APPROVED** [review]
Approved!

**Pull request merged**